kore/src/Kore/Internal/SideCondition.hs

```diff
@@ -208,3 +351,225 @@ toRepresentation
 toRepresentation =
     mkRepresentation
     . mapVariables @_ @VariableName (pure toVariableName)
+
+{- | Looks up the term in the table of replacements.
+ -}
+replaceTerm
+    :: InternalVariable variable
+    => SideCondition variable
+    -> TermLike variable
+    -> Maybe (TermLike variable)
+replaceTerm SideCondition { replacements } original =
+    HashMap.lookup original replacements
+
+{- | If the term isn't a key in the table of replacements
+then it cannot be replaced.
+ -}
+cannotReplaceTerm
+    :: InternalVariable variable
+    => SideCondition variable
+    -> TermLike variable
+    -> Bool
+cannotReplaceTerm SideCondition { replacements } term =
+    HashMap.lookup term replacements & isNothing
+
+data DoubleMap variable = DoubleMap
```

Comment:
This isn't a generic type, so it should not have a generic name like `DoubleMap`. Let's call it `Assumptions` instead.